src/test/groovy/com/commercehub/gradle/plugin/avro/OptionsFunctionalSpec.groovy

```diff
@@ -198,4 +201,34 @@ class OptionsFunctionalSpec extends FunctionalSpec {
         taskInfoAbsent || result.task(":generateAvroJava").outcome == FAILED
         result.output.contains("Invalid fieldVisibility 'badValue'.  Value values are: [PUBLIC, PUBLIC_DEPRECATED, PRIVATE]")
     }
+
+    @Unroll
+    def "supports configuring enableDecimalLogicalType to #enableDecimalLogicalType"() {
+        given:
+        copyResource("user.avsc", avroDir)
+        buildFile << """
+        |avro {
+        |    enableDecimalLogicalType = $enableDecimalLogicalType
+        |}
+        |""".stripMargin()
+
+        when:
+        def result = run("generateAvroJava")
+
+        then: "the task succeeds"
+        taskInfoAbsent || result.task(":generateAvroJava").outcome == SUCCESS
+        def content = projectFile("build/generated-main-avro-java/example/avro/User.java").text
+
+        and: "the specified enableDecimalLogicalType is used"
+        content.contains("public void setSalary(${BigDecimal.name} value)") == expectedPresent
```

Comment:
I'd like for this to assert on the expected content both with the flag enabled or disabled.  That is, if the flag is disabled, it should assert that `ByteBuffer` is used.

Reply:
Ok, will adjust  the code